x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -258,23 +250,35 @@ static String field(AggregateFunction af) {
             }
             return field.name();
         }
-        if (arg instanceof Literal) {
-            return String.valueOf(((Literal) arg).value());
-        }
         throw new SqlIllegalArgumentException("Does not know how to convert argument {} for function {}", arg.nodeString(),
                                               af.nodeString());
     }
 
-    private static String topAggsField(AggregateFunction af, Expression e) {
+    private static Object topAggsFieldOrScript(AggregateFunction af, Expression e) {
```

Comment:
This can be moved under the TopHit agg class

Reply:
Consolidated into one method.